src/modules/dashboard/transformers.ts

```diff
@@ -116,3 +122,36 @@ export const getActionsListData = (
   });
   return formattedActions;
 };
+
+export const getEventsListData = (
+  unformattedEvents?: SubgraphEvents,
+): FormattedEvent[] =>
+  unformattedEvents?.events
+    .map((event) => {
+      if (event) {
+        const {
+          id,
+          associatedColony: { colonyAddress },
+          transaction: {
+            hash,
+            block: { timestamp },
+          },
+          name,
+          args,
+        } = event;
+        const values = JSON.parse(args);
+        return {
+          id,
+          agent: values.agent ? createAddress(values.agent) : null,
+          eventName: formatEventName(name),
+          transactionHash: hash,
+          colonyAddress: createAddress(colonyAddress),
+          createdAt: new Date(parseInt(`${timestamp}000`, 10)),
+          values,
+          displayValues: args,
+          fromDomain: values?.domainId || null,
+        };
+      }
+      return undefined;
+    })
```

Comment:
if you would use reduce, you wouldn't need to return undefined and then filter undefined

Reply:
Actually that's a fair point